plugins/fpc/fu-fpc-device.c

```diff
@@ -292,22 +297,50 @@ fu_fpc_device_check_dfu_status_cb(FuDevice *device, gpointer user_data, GError *
 	return TRUE;
 }
 
+static gboolean
+fu_fpc_device_clear_status_cb(FuDevice *device, gpointer user_data, GError **error)
+{
+	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(device));
+	FuFpcDevice *self = FU_FPC_DEVICE(device);
+	g_autoptr(GError) error_local = NULL;
+
+	//force to reboot GUSBdevice
+	if (self->need_reset == TRUE && !g_usb_device_reset(usb_device, &error_local)) {
+		g_usb_device_claim_interface (usb_device, 0, 0, &error_local);
+		g_prefix_error(&error_local, "failed to force-reset device: ");
+		g_propagate_error(error, g_steal_pointer(&error_local));
+		return FALSE;
+	}
```

Comment:
the error handling here is funny. I'd do something like:

	if (self->need_reset) {
		if (!g_usb_device_reset(usb_device, error)) {
			g_prefix_error(error, "failed to force-reset device: ");
			return FALSE;
		}
		if (!g_usb_device_claim_interface(usb_device, 0, G_USB_DEVICE_CLAIM_INTERFACE_NONE, error)) {
			g_prefix_error(error, "failed to claim interface after reset: ");
			return FALSE;
		}
		self->need_reset = FALSE;
	}

Reply:
OK